### PR TITLE
Ensure AnnotationLiterals for java.lang annotations don't throw security exceptions

### LINKED ISSUE #4659: SecurityException for injection point with @Deprecated
**Describe the bug**

I have an injected config property:
```
    @Inject
    @ConfigProperty(name = "foo", defaultValue = "0")
    int foo;
```
If I add `@Deprecated` then I get a security exception as follows:

```
org.junit.jupiter.api.extension.TestInstantiationException: TestInstanceFactory [io.quarkus.test.junit.QuarkusTestExtension] failed to instantiate test class [test.Test]

	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeTestInstanceFactory(ClassBasedTestDescriptor.java:300)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateTestClass(ClassBasedTestDescriptor.java:275)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateTestClass(ClassTestDescriptor.java:77)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateAndPostProcessTestInstance(ClassBasedTestDescriptor.java:258)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$2(ClassBasedTestDescriptor.java:252)
	at java.util.Optional.orElseGet(Optional.java:267)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$3(ClassBasedTestDescriptor.java:251)
	at org.junit.jupiter.engine.execution.TestInstancesProvider.getTestInstances(TestInstancesProvider.java:29)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$prepare$0(TestMethodTestDescriptor.java:106)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:105)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:69)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$prepare$1(NodeTestTask.java:107)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.prepare(NodeTestTask.java:107)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:75)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:229)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:197)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:191)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:69)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.lang.ExceptionInInitializerError
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:134)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:229)
	at io.quarkus.test.junit.QuarkusTestExtension.createTestInstance(QuarkusTestExtension.java:348)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeTestInstanceFactory(ClassBasedTestDescriptor.java:285)
	... 47 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl1.<clinit>(ApplicationImpl1.zig:469)
	... 56 more
Caused by: java.lang.SecurityException: Prohibited package name: java.lang
	at java.lang.ClassLoader.preDefineClass(ClassLoader.java:662)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:761)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at example.Example_Bean.<init>(Example_Bean.zig:1095)
	at io.quarkus.arc.setup.Default_ComponentsProvider.getComponents(Default_ComponentsProvider.zig:1463)
	at io.quarkus.arc.ArcContainerImpl.<init>(ArcContainerImpl.java:86)
	at io.quarkus.arc.Arc.initialize(Arc.java:18)
	at io.quarkus.arc.runtime.ArcRecorder.getContainer(ArcRecorder.java:33)
	at io.quarkus.deployment.steps.ArcProcessor$generateResources26.deploy_0(ArcProcessor$generateResources26.zig:72)
	at io.quarkus.deployment.steps.ArcProcessor$generateResources26.deploy(ArcProcessor$generateResources26.zig:36)
	at io.quarkus.runner.ApplicationImpl1.<clinit>(ApplicationImpl1.zig:391)
	... 56 more
```

**Expected behavior**

No exception.

**Actual behavior**
(Describe the actual behavior clearly and concisely.)

Exception as above.

**To Reproduce**
1. See above.

**Configuration**
```properties
# Add your application.properties here, if applicable.
N/A
```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)
N/A

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: macOS Catalina
 - Output of `java -version`: 1.8.0_181
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 0.24.0

**Additional context**
(Add any other context about the problem here.)

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #4659

### CODE REVIEW

**Comment by geoand** [comment]
@mkouba I am not sure this needs to be taken care of anywhere else as well

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java [hunk 1]
```diff
@@ -1594,6 +1594,12 @@ static ResultHandle collectAnnotations(ClassOutput classOutput, ClassCreator bea
                         constructor.readStaticField(FieldDescriptor.of(InjectLiteral.class, "INSTANCE", InjectLiteral.class)));
             } else {
                 // Create annotation literal if needed
+
+                // we need to skip java lang annotations since the generated classes would end up in the java.lang package
+                // which the jvm does not allow
+                if (annotation.name().toString().startsWith("java.lang")) {
```
Comment (mkouba) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java]: I don't think we want to skip the literals but use a different package instead...

**Review by mkouba — CHANGES_REQUESTED** [review]

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java (hunk 1 above)
Comment (geoand) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java]: What package do you propose?

Comment (mkouba) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java]: We should probably use `io.quarkus.arc.processor.AbstractGenerator.DEFAULT_PACKAGE` (this is already used for client proxies).

Comment (geoand) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java]: Great thanks! I'll update the PR later on (or over the weekend).

Comment (mkouba) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java]: See also `io.quarkus.arc.processor.AnnotationLiteralGenerator.generatedSharedName()` and `io.quarkus.arc.processor.AnnotationLiteralGenerator.generatedLocalName()`.

Comment (geoand) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java]: PR updated per your suggestion @mkouba. Thanks!

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationLiteralGenerator.java [hunk 2]
```diff
@@ -252,8 +252,14 @@ static String componentType(MethodInfo method) {
     }
 
     static String generatedSharedName(DotName annotationName) {
+        // when the annotation is a java.lang annotation we need to use a different package in which to generate the literal
+        // otherwise a security exception will be thrown when the literal is loaded
+        String nameToUse = annotationName.toString().startsWith("java.lang")
```
Comment (mkouba) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationLiteralGenerator.java]: Looks good. Could we use the same logic for `generatedLocalName()` as well. It's not used for Quarkus but we should be consistent ;-).

Comment (geoand) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationLiteralGenerator.java]: Makes perfect sense! Done :)

**Review by mkouba — APPROVED** [review]

**Pull request merged**